ansible/playbook/hetzner-create-hcloud-server.yml

```diff
@@ -51,20 +51,21 @@
     - name: Delete cloud server
       shell: "hcloud server delete {{ vm_name }}"
       register: hcloud_delete_res
-      changed_when: "hcloud_delete_res.rc == 0 or ('server not found' in hcloud_delete_res.stderr)"
+      changed_when: "hcloud_delete_res.rc == 0"
       failed_when: "hcloud_delete_res.rc != 0 and ('server not found' not in hcloud_delete_res.stderr)"
-      when: "vm_delete is defined and vm_delete"
+      when: "vm_delete is defined and vm_delete | bool"
 
     - name: Delete cloud server key
       shell: "hcloud ssh-key delete snowdrop"
       register: hcloud_delete_key_res
       changed_when: "hcloud_delete_key_res.rc == 0"
-      when: "vm_delete is defined and vm_delete"
+      failed_when: "hcloud_delete_key_res.rc != 0 and ('SSH key not found' not in hcloud_delete_key_res.stderr)"
+      when: "vm_delete is defined and vm_delete | bool"
```

Comment:
Will it be possible with this change to use then `-e vm_delete=true` ? @jacobdotcosta

Reply:
Yes, I tried it and it works. 

```bash
$ ansible-playbook playbook/hetzner-create-hcloud-server.yml -e vm_name=myname -e salt_text=mysalt -e password_text=mypasswprd -e hetzner_context_name=mycontext -e vm_delete=true -e pass_store_dir="mypasswordstorelocation"
```

Reply:
Perfect. I ill then merge the PR.